src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -149,6 +149,21 @@ public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
             }
         }
 
+        internal override void AddSynthesizedAttributes(ModuleCompilationState compilationState, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            // Emit [Dynamic] on synthesized parameter symbols when the original parameter was dynamic 
+            // in order to facilitate debugging.  In the case the necessary attributes are missing 
+            // this is a no-op.  Emitting an error here, or when the original parameter was bound, would
+            // adversely effect the compilation or potentially change overload resolution.  
+            var compilation = this.DeclaringCompilation;
+            if (Type.ContainsDynamic() && 
+                compilation.HasDynamicEmitAttributes() &&
+                compilation.GetSpecialType(SpecialType.System_Boolean).SpecialType == SpecialType.System_Boolean)
```

Comment:
Because we are trying to answer the question whether we can use System.Boolean type and only use-site errors (or a lack of them) can give us the answer.